### PR TITLE
Design Updates, Error Messages, and Form Field Adjustments

### PR DESCRIPTION
**Pull request by nadezdatsygankova**
This PR enhances the sign-up form based on the latest Figma design specifications. 
Design Update

Figma Alignment: Updated the sign-up form to match the latest Figma design version.
Styling Enhancements: Improved the visual styling for inputs, buttons, and labels.

Form Field Modifications

Added Fields: Introduced input fields for First Name and Last Name.
Removed Field: Eliminated the Password Confirmation field to streamline the user registration process.

Error Message Implementation

Inline Error Messages: Added inline error messages for invalid inputs, providing immediate feedback to users.
Visual Feedback: Incorporated visual feedback for errors, such as red borders and error icons next to problematic fields.

Backend Update:

User Model: Updated the user model to include firstName and lastName fields.

Screenshots:

Figma Design:
Screenshot 2024-06-14 at 10 27 39 PM

Before Update

Screenshot 2024-06-14 at 10 13 04 PM

After Update 
[
Screenshot 2024-06-14 at 9 57 15 PM
](url)

Testing Instructions
Navigate to the Sign-Up Page: Confirm the form layout and styling match the Figma design.
Input Validation: Test the new fields (First Name, Last Name) for proper validation.
Error Handling: Verify error messages appear correctly when invalid data is entered.
Form Submission: Ensure the form is submitted correctly without the password confirmation field. 

Note. 
Please, during testing, disregard the console error related to App.jsx:33. This issue is unrelated to the sign-up form and is being addressed separately.

### CODE REVIEW

File: frontend/src/pages/Register/Register.jsx [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
 import styles from './Register.module.css';
+
 import RegisterForm from './RegisterForm';
 
 import { register } from '../../api';
-import FatherSonBlock from '../../components/FatherSonBlock';
+// import FatherSonBlock from '../../components/FatherSonBlock';
 import Header from '../../components/Header';
-import TextLink from '../../components/TextLink';
+// import TextLink from '../../components/TextLink';
```
Comment (AnnaSolovykh) [inline frontend/src/pages/Register/Register.jsx]: Will we need commented out code in the future?

File: frontend/src/pages/Register/Register.module.css [hunk 2]
```diff
@@ -1,12 +1,166 @@
...
+.firstNameLabel,
+.LastNameLabel {
+  font-family: Roboto;
+  font-weight: 600;
+  font-size: 16px;
+  line-height: 15.36px;
+  letter-spacing: 2%;
+  color: #081821;
+}
+.EmailLabel,
```
Comment (AnnaSolovykh) [inline frontend/src/pages/Register/Register.module.css]: Why is PascalCase used for these classnames? Is it occasionally or on purpose? If occasionally, let's follow the common pattern and use camelCase.

File: frontend/src/pages/Register/Register.module.css [hunk 3]
```diff
@@ -1,12 +1,166 @@
...
+  text-align: center;
+  color: #fcfcfc;
+}
+.customButton {
+  margin-top: 24px;
+  width: 442px;
+  height: 56px;
+  /* Add other button styles here */
+}
+.SocialButton {
```
Comment (AnnaSolovykh) [inline frontend/src/pages/Register/Register.module.css]: Why is PascalCase used for these classnames? Is it occasionally or on purpose? If occasionally, let's follow the common pattern and use camelCase.

File: frontend/src/pages/Register/Register.module.css [hunk 4]
```diff
@@ -1,12 +1,166 @@
...
 
 .signUpText {
   font-size: 18px;
   line-height: 21px;
   text-align: center;
-}
\ No newline at end of file
+  color: aliceblue;
+}
+/* In Register.module.css or your preferred CSS file */
```
Comment (AnnaSolovykh) [inline frontend/src/pages/Register/Register.module.css]: This is a bit strange comment :) Comments should be added to explain some difficult patterns to the project's new-comers. 
I don't think we need this comment :)

File: frontend/src/pages/Register/Register.module.css [hunk 5]
```diff
@@ -1,12 +1,166 @@
...
+  height: 1px; /* Set the height */
+  background-color: #454848; /* Set the color */
+}
+
+.dashLine:before {
+  left: -220px; /* Adjust the position of the first dash line */
+}
+
+.dashLine:after {
+  right: -220px; /* Adjust the position of the second dash line */
```
Comment (AnnaSolovykh) [inline frontend/src/pages/Register/Register.module.css]: Is making these comments justified? Seems like they are explaining basic stuff.

File: frontend/src/pages/Register/Register.module.css [hunk 6]
```diff
@@ -1,12 +1,166 @@
...
+  color: #454848;
+}
+.alreadyMember {
+  text-align: center;
+  margin-top: 20px;
+}
+.loginLink {
+  cursor: pointer;
+  color: #f08324; /* Set color to white */
+  text-decoration: underline; /* Add underline */
```
Comment (AnnaSolovykh) [inline frontend/src/pages/Register/Register.module.css]: The same. The color isn't even white. 
Screenshot 2024-06-17 at 11 33 59

**Review by AnnaSolovykh — CHANGES_REQUESTED** [review]
Well done! Everything looks really nice, good job! 
Though there are a few minor issues to look into. 
Also Log In button under the register form does not work.
Screenshot 2024-06-17 at 11 35 45

File: frontend/src/pages/Register/Register.jsx (hunk 1 above)
Comment (VrushaliHaldankar01) [reply frontend/src/pages/Register/Register.jsx]: Hii all the commented lines from the code has been removed.

File: frontend/src/pages/Register/Register.module.css (hunk 2 above)
Comment (VrushaliHaldankar01) [reply frontend/src/pages/Register/Register.module.css]: The class name has been changed

**Comment by AnnaSolovykh** [comment]
@VrushaliHaldankar01 thanks, Vrushali! Looks good to me!

We also need to fix log in link on the register page. Could you be so kind to fix it too?

**Comment by AnnaSolovykh** [comment]
@nadezdatsygankova @KatePolyakov @VrushaliHaldankar-01 look, also I don't see this error in the dashboard branch - I guess it should be fixed before merging. Who can take it?
Screenshot 2024-06-17 at 20 12 05

**Comment by nadezdatsygankova** [comment]
> Well done! Everything looks really nice, good job! Though there are a few minor issues to look into. Also Log In button under the register form does not work. Screenshot 2024-06-17 at 11 35 45

Thank you for catching these errors. 
I've fixed the link in the 'login in' button and removed commented lines.

**Comment by nadezdatsygankova** [comment]
> @VrushaliHaldankar01 thanks, Vrushali! Looks good to me!
> 
> We also need to fix log in link on the register page. Could you be so kind to fix it too?

I've fixed log in link

**Comment by nadezdatsygankova** [comment]
> @nadezdatsygankova @KatePolyakov @VrushaliHaldankar-01 look, also I don't see this error in the dashboard branch - I guess it should be fixed before merging. Who can take it? Screenshot 2024-06-17 at 20 12 05

I've fixed it. Thank you

**Review by AnnaSolovykh — APPROVED** [review]

**Review by KatePolyakov — APPROVED** [review]
Well Done Ladies

**Review by S-chahine — COMMENTED** [review]
Great job, ladies! I have tested the form submission process, and everything is working fine overall. However, I encountered a few errors that I'm not sure about:

Sign-Up Button Error:
When I click the Sign-Up button, I get the following error:
image_720 (1)


Page Load Error:
Once the page is loaded, this error appears in the console:


image_480

Login Button Error:
When I click the Login button, this error message appears:
image_720 (1)

Could you please take a look at these issues? Thanks!

**Comment by AnnaSolovykh** [comment]
> Great job, ladies! I have tested the form submission process, and everything is working fine overall. However, I encountered a few errors that I'm not sure about:
> 
> Sign-Up Button Error: When I click the Sign-Up button, I get the following error: image_720 (1)
> 
> Page Load Error: Once the page is loaded, this error appears in the console:
> 
> image_480
> 
> Login Button Error: When I click the Login button, this error message appears: image_720 (1)
> 
> Could you please take a look at these issues? Thanks!

I have put these errors in the separate ticket  - https://github.com/users/JULIERAJ/projects/2/views/1?pane=issue&itemId=67482131  We will need a volunteer tomorrow to fix them :)

**Review by AnnaSolovykh — APPROVED** [review]

**Review by Skyline857 — APPROVED** [review]
Amazing job, ladies! I've tested the submission form. Everything looks great overall! However, when I try to click on Sign- Up button, I see these type of errors in my console

registertest

**Comment by AnnaSolovykh** [comment]
> Amazing job, ladies! I've tested the submission form. Everything looks great overall! However, when I try to click on Sign- Up button, I see these type of errors in my console
> 
> registertest

This is fine - we are not connected to the api yet.

**Pull request merged**